src/App.jsx

```diff
@@ -10,6 +10,8 @@ import { generateToken } from '@the-collab-lab/shopping-list-utils';
 export function App() {
 	const navigateTo = useNavigate();
 	const [data, setData] = useState([]);
+	const [loading, setLoading] = useState(true);
```

Comment:
niceeeee, adding a loader!

Reply:
> Great work on this issue! I liked that you added the loader mechanism and set it in the streamListItems function and pass it down! I was wondering if we should do anything to guard the data.length in the conditional render but your loader should serve as that guard -- was that your inspiration behind adding it? Would love to know your thought process!

Yessss! Would love to speak to that in today's demo! We had finished up the acceptance criteria, but noticed that every time we hit the page we saw a quick 1-second glance of the search functionality (which we had hid behind some conditional rendering, as you noted previously)! Our thought was that the page was loading before it knew whether or not there was data in there (which is the base of the logic behind that render). How do you stop a component from rendering before it knows if it should? Loading state! When we added that, the problem was solved. 🙌